test/unit/requestWrapper.test.js

```diff
@@ -1,6 +1,279 @@
+        url:
+          'https://example.ibm.com/v1/environments/environment-id/configurations/configuration-id',
+        headers: {
+          'test-header': 'test-header-value',
+        },
+        responseType: 'jsonp',
+      },
+    };
+
+    axios.mockResolvedValue('res');
+
+    sendRequest(parameters, (err, body, res) => {
+      // assert results
+      expect(axios.mock.calls[0][0].data).toEqual('post=body');
+      expect(axios.mock.calls[0][0].url).toEqual(
+        'https://example.ibm.com/v1/environments/environment-id/configurations/configuration-id'
+      );
+      expect(axios.mock.calls[0][0].headers).toEqual({
+        'Accept-Encoding': 'gzip',
+        'test-header': 'test-header-value',
+      });
+      expect(axios.mock.calls[0][0].httpsAgent.options.rejectUnauthorized).toEqual(
+        parameters.defaultOptions.rejectUnauthorized
+      );
+      expect(axios.mock.calls[0][0].method).toEqual(parameters.defaultOptions.method);
+      expect(axios.mock.calls[0][0].responseType).toEqual(parameters.defaultOptions.responseType);
+      expect(res).toEqual('res');
+      expect(axios.mock.calls.length).toBe(1);
+      done();
+    });
+  });
+
+  it('should call formatError if request failed', done => {
+    const parameters = {
+      defaultOptions: {
+        body: 'post=body',
+        formData: '',
+        qs: {},
+        method: 'POST',
+        rejectUnauthorized: true,
+        url:
+          'https://example.ibm.com/v1/environments/environment-id/configurations/configuration-id',
+        headers: {
+          'test-header': 'test-header-value',
+        },
+        responseType: 'json',
+      },
+    };
+
+    axios.mockRejectedValue('error');
+
+    sendRequest(parameters, (err, body, res) => {
+      // assert results
+      expect(err).toEqual(expect.anything());
+      expect(body).toBeUndefined();
+      expect(res).toBeUndefined();
+      done();
+    });
+  });
+
+  it('should send a request with overwritten parameters', done => {
+    const parameters = {
+      defaultOptions: {
+        formData: '',
+        qs: {
+          version: '2017-10-15',
+        },
+        method: 'POST',
+        rejectUnauthorized: true,
+        url: 'https://example.ibm.com',
+        headers: {
+          'test-header': 'test-header-value',
+        },
+      },
+      options: {
+        url: '/v1/environments/{environment_id}/configurations/{configuration_id}',
+        path: {
+          environment_id: 'environment-id',
+          configuration_id: 'configuration-id',
+        },
+        rejectUnauthorized: false,
+        method: 'GET',
+        qs: {
+          version: '2018-10-15',
+          array_style: ['a', 'b'],
+        },
+        headers: {
+          'test-header': 'override-header-value',
+          'add-header': 'add-header-value',
+        },
+      },
+    };
+
+    axios.mockImplementation(requestParams => {
+      requestParams.paramsSerializer(requestParams.params);
```

Comment:
I see what you're doing here, but I'm not sure why it's necessary. Will you add a comment explaining why this line is needed?